### PR TITLE
BattleSys_ -> BattleSystem_ + Battle overlay progress

### PR DESCRIPTION
**Pull request by adrienntindall**

### CODE REVIEW

File: src/battle/overlay_12_0224E4FC.c [hunk 1]
```diff
@@ -2883,3 +2884,414 @@ BOOL CheckMoveCallsOtherMove(u16 moveNo) {
...
+    return ret;
+}
+
+u8 GetBattlerAbility(BATTLECONTEXT *ctx, int battlerId) {
+    if ((ctx->battleMons[battlerId].moveEffectFlags & MOVE_EFFECT_GASTRO_ACID) && ctx->battleMons[battlerId].ability != ABILITY_MULTITYPE) {
+        return 0;
+    } else if ((ctx->fieldCondition & FIELD_CONDITION_GRAVITY) && ctx->battleMons[battlerId].ability == ABILITY_LEVITATE) {
+        return 0;
+    } else if ((ctx->battleMons[battlerId].moveEffectFlags & MOVE_EFFECT_INGRAIN) && ctx->battleMons[battlerId].ability == ABILITY_LEVITATE) {
+        return 0;
```
Comment (red031000) [inline src/battle/overlay_12_0224E4FC.c:2929]: do we want a constant for invalid/no ability?

File: src/battle/overlay_12_0224E4FC.c [hunk 2]
```diff
@@ -2883,3 +2884,414 @@ BOOL CheckMoveCallsOtherMove(u16 moveNo) {
...
+    }
+    if (cnt >= cntMax) {
+        ret = TRUE;
+    }
+    return ret;
+}
+
+//Weird because it returns FALSE if you *can* escape, TRUE if you *can't*
+//FIXME: Function needs a more descriptive name
+BOOL CanEscape(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId, BATTLEMSG *msg) {
```
Comment (red031000) [inline src/battle/overlay_12_0224E4FC.c:3000]: CanNotEscape?

**Review by red031000 — CHANGES_REQUESTED** [review]

File: src/battle/overlay_12_0224E4FC.c (hunk 1 above)
Comment (adrienntindall) [reply src/battle/overlay_12_0224E4FC.c:2929]: just checked and we already have one, will put the const here

File: src/battle/overlay_12_0224E4FC.c (hunk 2 above)
Comment (adrienntindall) [reply src/battle/overlay_12_0224E4FC.c:3000]: why not

**Review by red031000 — APPROVED** [review]

**Pull request merged**